proposals/LP0004-project-governance.md

```diff
@@ -0,0 +1,393 @@
+
+_Area teams_ are not intended to be direction setters or primary maintainers of their areas, although individuals on an _area team_ may fulfill that role separately. The _area team’s_ role is as a steward and moderator ensuring the health and smooth operation of the area.
+
+Each _area team_ will have 5 members elected by the _voting contributors_. Candidates for _area teams_ must be a _voting contributor_ and self-nominated. An individual cannot serve on two _area teams_.
+
+Members of an _area team_ are elected for 1 year terms. Individuals are encouraged to not run for more than two consecutive terms. To encourage turnover, incumbent members of an _area team_ must place in the top 4 candidates during an election. The fifth spot will always go to a non-incumbent candidate.
+
+
+#### Initial Area Teams
+
+With this proposal a group of initial _area teams_ will be formed with defined domain areas. The members of each _area team_ will be defined by an election held after the adoption of the proposal. The initial _area teams_ proposed below are based on statistics generated from the LLVM project repository (See Appendix 2 for applicable scripts).
+
+In dividing the LLVM project into _area teams_, this proposal seeks to break up areas of the project based on contribution volume and number of active contributors. This proposal assumes that areas of the project with more unique contributors are more likely to have contributor disagreements, and that areas with more highly active contributors will be more likely to have diverse _area team_ formations.
+
+This proposal puts a possible set of _area teams_ to form with the specified areas of ownership. One notable point is that LLVM (`:/llvm`) had 945 unique authors contribute over the 12 month period used for statistics. This is far more than any other part of the project, and it justified splitting LLVM into two teams, which are separated as the _LLVM Backend_ and _LLVM Middle-end_. Clang had 706 unique authors, which is also significantly higher than other sub-projects. To represent different parts of the Clang community the _Clang_ and _Clang Tooling_ teams are separated below.
+
+* LLVM Backend (CodeGen, MC, Target)
+* LLVM Middle-end (Everything not covered by LLVM Backend)
+* Clang (All the `clang` sub-project except tooling and format libraries and tools)
+* Clang Tooling (`clang-tools-extra` and the tooling and format libraries from `clang`)
+* C/C++ Runtime Libraries (`libcxx`, `libc`, `libunwind`, `libcxxabi`)
```

Comment:
C/C++ front-end feels like its own category but does not break down nicely based on directories. From a screening, code review and commits perspective this is clearly its own area.

Contributions here also are not clearly measured directly in the llvm repo but can involve WG21 and WG14 work including proposals, defect reports and participation in EWG, CWG meeting.

Reply:
Clang not splitting by directory isn't entirely unique. LLVM doesn't split well by directory either. LLVM and Clang get enough contributor attention that I thought we needed more than one _area team_ responsible for providing oversight to spread the load around. I'm open for suggestions about how to better make that split. My idea was basically "the language compiler" and "the language tools" recognizing that the tools rely on the compiler and there is some blurry overlap. This does need to be better specified.

I think we need to explicitly exclude participation outside the LLVM project. This isn't "C Language Governance" or "C-based Language Governance", it's "LLVM Project Governance".

I do agree that participation in the LLVM associated working groups matters, but I'm unsure if someone attending a working group meeting but not contributing to LLVM/Clang in any other way should garner voting rights.

Reply:
Placing libc++/libc++abi/libunwind together makes a lot of sense. libc has a very different sub-community, so it can be moved into its own group?

lld/bolt and compiler-rt/openmp are also very different. Let's use lld as an example. If each of ELF/MachO/COFF/MinGW/wasm ports needs a team member, then there will be at least 5 members.

Reply:
Please make libc distinct from the rest of the runtimes.